src/main/java/WayofTime/bloodmagic/util/helper/RitualHelper.java

```diff
@@ -117,4 +119,135 @@ else if (tile != null && tile.hasCapability(RUNE_CAPABILITY, null)) {
+        }
+    }
+
+    public static boolean createRitual(World world, BlockPos pos, EnumFacing direction, Ritual ritual, boolean safe) {
+
+        List<RitualComponent> components = Lists.newArrayList();
+        ritual.gatherComponents(components::add);
+
+        if (abortConstruction(world, pos, direction, safe, components)) return false;
+
+        IBlockState mrs = RegistrarBloodMagicBlocks.RITUAL_CONTROLLER.getDefaultState();
+        world.setBlockState(pos, mrs);
+
+        setRitualStones(direction, world, pos, components);
+        return true;
+    }
+
+    public static boolean abortConstruction(World world, BlockPos pos, EnumFacing direction, boolean safe, List<RitualComponent> components) {
+        //TODO: can be optimized to check only for the first and last component if every ritual has those at the highest and lowest y-level respectivly.
+        for (RitualComponent component : components) {
+            BlockPos offset = component.getOffset(direction);
+            BlockPos newPos = pos.add(offset);
+            if (world.isOutsideBuildHeight(newPos) || (safe && !world.isAirBlock(newPos)))
+                return true;
+        }
+        return false;
+    }
+
+    public static void removeRitualFromRuins(TileMasterRitualStone tile) {
+        Pair<Ritual, EnumFacing> pair = getRitualFromRuins(tile);
+        Ritual ritual = pair.getKey();
+        EnumFacing direction = pair.getValue();
+        BlockPos pos = tile.getPos();
+        World world = tile.getWorld();
+        List<RitualComponent> components = Lists.newArrayList();
+        ritual.gatherComponents(components::add);
+        removeRitualStones(direction, pos, world, components);
+    }
+
+    public static boolean repairRitualFromRuins(TileMasterRitualStone tile, boolean safe) {
+        Pair<Ritual, EnumFacing> pair = getRitualFromRuins(tile);
+        Ritual ritual = pair.getKey();
+        EnumFacing direction = pair.getValue();
+        World world = tile.getWorld();
+        BlockPos pos = tile.getPos();
+
+        List<RitualComponent> components = Lists.newArrayList();
+        ritual.gatherComponents(components::add);
+
+        if (abortConstruction(world, pos, direction, safe, components)) return false;
+
+        setRitualStones(direction, world, pos, components);
+        return true;
+    }
+
+    public static void setRitualStones(EnumFacing direction, World world, BlockPos pos, List<RitualComponent> gatheredComponents) {
+        for (RitualComponent component : gatheredComponents) {
+            BlockPos offset = component.getOffset(direction);
+            BlockPos newPos = pos.add(offset);
+            int meta = component.getRuneType().ordinal();
+            IBlockState newState = RegistrarBloodMagicBlocks.RITUAL_STONE.getStateFromMeta(meta);
+            world.setBlockState(newPos, newState);
+        }
+    }
+
+
+    public static Pair<Ritual, EnumFacing> getRitualFromRuins(TileMasterRitualStone tile) {
```

Comment:
Again... the MRS already knows this information.

Reply:
This is also supposed to be able to be used on "Rituals" that have never been activated.